### PR TITLE
feat(JOML)!: migrate `BlockComponent`

### LINKED ISSUE #4430: Multiple systems can write to and change a BlockComponent
BlockComponents can either be persisted where their components are saved or live in a temporary half state where stuff can be set by another system and at the start of the next update loop these temporary entities are cleared and are destroyed. its very likely that the position stored can be modified by 1 system and read back out by another system in the event loop. This can be unintentional i.e the position is used in some other calculation that updates the internal position of the BlockComponent.  I think this can introduces undefined behavior where one system expects a block at a given position and reads out something else. from my understanding the position used by BlockComponent does not actually affect the block in the world. i.e move it in any sort of way.

I'm kind of concerned this would introduce undefined behavior.

### PR DESCRIPTION
**Pull request by skaldarnar**
Dependant module PRs are linked back to this PR. To test it out, checkout the branch `joml/migrate-BlockComponent` on all those modules.

```sh
# in Terasology workspace root

git checkout joml/migrate-BlockComponent

for m in ItemRendering Explosives Furnishings ItemPipes AdditionalItemPipes Rails Signalling AdditionalRails AdventureAssets AlterationEffects ClimateConditions GrowingFlora MultiBlock Behaviors Fences ComputerMonitors Machines ManualLabor Exoplanet FunnyBlocks KComputers MasterOfOreon Sample Scenario WoodAndStone SimpleFarming; do
    git -C $(pwd)/modules/$m fetch;
    git -C $(pwd)/modules/$m checkout joml/migrate-BlockComponent;
done
```

- [ ] https://github.com/Terasology/ItemRendering/pull/18
- [ ] https://github.com/Terasology/Explosives/pull/4
- [ ] https://github.com/Terasology/Furnishings/pull/9
- [ ] https://github.com/Terasology/ItemPipes/pull/32
- [ ] https://github.com/Terasology/AdditionalItemPipes/pull/21
- [ ] https://github.com/Terasology/Rails/pull/74
- [ ] https://github.com/Terasology/Signalling/pull/28
- [ ] https://github.com/Terasology/AdditionalRails/pull/46
- [ ] https://github.com/Terasology/AdventureAssets/pull/36
- [ ] https://github.com/Terasology/AlterationEffects/pull/26
- [ ] https://github.com/Terasology/ClimateConditions/pull/46
- [ ] https://github.com/Terasology/GrowingFlora/pull/30
- [ ] https://github.com/Terasology/MultiBlock/pull/22
- [ ] https://github.com/Terasology/Behaviors/pull/68
- [ ] https://github.com/Terasology/Fences/pull/16
- [ ] https://github.com/Terasology/ComputerMonitors/pull/8
- [ ] https://github.com/Terasology/Machines/pull/50
- [ ] https://github.com/Terasology/ManualLabor/pull/42
- [ ] https://github.com/Terasology/Exoplanet/pull/20
- [ ] https://github.com/Terasology/FunnyBlocks/pull/23
- [ ] https://github.com/Terasology/KComputers/pull/10
- [ ] https://github.com/Terasology/MasterOfOreon/pull/93
- [ ] https://github.com/Terasology/Sample/pull/122
- [ ] https://github.com/Terasology/Scenario/pull/56
- [ ] https://github.com/Terasology/WoodAndStone/pull/72
- [ ] https://github.com/Terasology/SimpleFarming/pull/107

Contributes to #3832
Resolves #4430

### CODE REVIEW

File: engine/src/main/java/org/terasology/world/block/BlockComponent.java [hunk 1]
```diff
@@ -15,90 +15,68 @@
...
-     * @deprecated This is scheduled for removal in an upcoming version method will be replaced with JOML implementation
-     *     {@link #getPosition(org.joml.Vector3i)}.
+     * Get an immutable view on the current position.
+     *
+     * Note: the vector may change when the position on this component is updated. If the position information is to be
+     * stored you should use {@link #getPosition(Vector3i)} instead.
      */
-    @Deprecated
-    public Vector3i getPosition() {
+    public Vector3ic getPosition() {
```
Comment (pollend) [inline engine/src/main/java/org/terasology/world/block/BlockComponent.java:55]: I really wanted to do this.

Comment (skaldarnar) [reply engine/src/main/java/org/terasology/world/block/BlockComponent.java:55]: I just did it ^^ :nerd_face:

File: engine/src/main/java/org/terasology/world/block/BlockComponent.java [hunk 2]
```diff
@@ -15,90 +15,68 @@
...
+        position.set(pos);
+    }
+
     /**
      * set the position of the {@link BlockComponent}
      *
      * @param pos position to set
      */
-    public void setPosition(org.joml.Vector3i pos) {
+    public void setPosition(org.terasology.math.geom.Vector3i pos) {
```
Comment (jdrueckert) [inline engine/src/main/java/org/terasology/world/block/BlockComponent.java:79]: Is this still used?

Comment (skaldarnar) [reply engine/src/main/java/org/terasology/world/block/BlockComponent.java]: nope, removed in 2f9afc4

**Review by jdrueckert — APPROVED** [review]

**Pull request merged**